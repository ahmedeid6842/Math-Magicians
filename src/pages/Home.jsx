import React from 'react';
import '../styles/home.css';

const Home = () => (
  <>
    <section className="container home-container">
      <h2>Welcome to Math Magicians!</h2>
      <p>
        Math Magicians is a website that provides a simple calculator for basic
        arithmetic operations, as well as a random quote generator to inspire
        your mathematical thinking. Our goal is to make math fun and accessible
        for everyone, regardless of their background or experience level.
      </p>
      <p>
        Whether you&apos;re a student looking to improve your math skills, a teacher
        searching for new ideas and resources, or simply someone who enjoys
        playing with numbers, Math Magicians has something for you. Try our
        calculator to solve simple or complex equations, or explore our quote
        generator to discover insightful and inspiring quotes.
      </p>
    </section>
  </>
);

export default Home;
