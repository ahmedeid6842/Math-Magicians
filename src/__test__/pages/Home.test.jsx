import React from 'react';
import { render } from '@testing-library/react';
import Home from '../../pages/Home';

describe('Home', () => {
  it('renders correctly', () => {
    const { container } = render(<Home />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('displays the welcome message', () => {
    const { getByText } = render(<Home />);
    expect(getByText(/welcome to math magicians!/i)).toBeInTheDocument();
  });

  it('displays the description', () => {
    const { getByText } = render(<Home />);
    expect(getByText(/math magicians is a website/i)).toBeInTheDocument();
  });
});
