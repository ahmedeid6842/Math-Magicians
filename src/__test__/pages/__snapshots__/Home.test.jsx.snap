// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<section
  class="container home-container"
>
  <h2>
    Welcome to Math Magicians!
  </h2>
  <p>
    Math Magicians is a website that provides a simple calculator for basic arithmetic operations, as well as a random quote generator to inspire your mathematical thinking. Our goal is to make math fun and accessible for everyone, regardless of their background or experience level.
  </p>
  <p>
    Whether you're a student looking to improve your math skills, a teacher searching for new ideas and resources, or simply someone who enjoys playing with numbers, Math Magicians has something for you. Try our calculator to solve simple or complex equations, or explore our quote generator to discover insightful and inspiring quotes.
  </p>
</section>
`;
